Spigot-Server-Patches/0673-Add-PlayerGlassBottleFillEvent.patch

```diff
@@ -0,0 +1,138 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Dmitry Sidorov <jonmagon@gmail.com>
+Date: Tue, 2 Feb 2021 19:53:04 +0300
+Subject: [PATCH] Add PlayerGlassBottleFillEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/BlockBeehive.java b/src/main/java/net/minecraft/server/BlockBeehive.java
+index eda45e7f2f824c42653d6c545172efc7c96ef332..50956f67e9d71c5cc5c887c1fd0c580629179336 100644
+--- a/src/main/java/net/minecraft/server/BlockBeehive.java
++++ b/src/main/java/net/minecraft/server/BlockBeehive.java
+@@ -1,6 +1,7 @@
+ package net.minecraft.server;
+ 
+ import io.papermc.paper.event.block.PlayerShearBlockEvent; // Paper - PlayerShearBlockEvent namespace conflicts
++import io.papermc.paper.event.player.PlayerGlassBottleFillEvent; // Paper
+ 
+ import java.util.Iterator;
+ import java.util.List;
+@@ -94,13 +95,30 @@ public class BlockBeehive extends BlockTileEntity {
+                 });
+                 flag = true;
+             } else if (itemstack.getItem() == Items.GLASS_BOTTLE) {
++                // Paper start - PlayerGlassBottleFillEvent
++                PlayerGlassBottleFillEvent event = new PlayerGlassBottleFillEvent((org.bukkit.entity.Player) entityhuman.getBukkitEntity(), org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(itemstack), org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(new ItemStack(Items.HONEY_BOTTLE)));
```

Comment:
`asBukkitCopy` -> `asCraftMirror` as per my last comment

```suggestion
+                PlayerGlassBottleFillEvent event = new PlayerGlassBottleFillEvent((org.bukkit.entity.Player) entityhuman.getBukkitEntity(), org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemstack), org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(new ItemStack(Items.HONEY_BOTTLE)));
```

Reply:
Also, for the HoneyBottle, just create a bukkit itemstack, no point in making a NMS one if you're going to copy it anyway